external/vcm/tests/test__preloaded.py

```diff
@@ -0,0 +1,88 @@
+from typing import Sequence, Tuple, Dict, Any
+import random
+import vcm
+import pytest
+import time
+
+
+class CallRecord:
+    def __init__(self, args, kwargs):
+        self.args: Tuple[Any] = args
+        self.kwargs: Dict[str, Any] = kwargs
+
+
+class DummyFunction:
+    def __init__(self):
+        self.call_history: Sequence[CallRecord] = []
+
+    def __call__(self, *args, **kwargs) -> CallRecord:
+        self.call_history.append(CallRecord(args, kwargs))
+        return CallRecord(args, kwargs)
+
+
+@pytest.fixture(params=[0, 1, 5])
+def n_filenames(request):
+    return request.param
+
+
+@pytest.fixture
+def filenames(n_filenames):
+    return [f"{i}.ext" for i in range(n_filenames)]
+
+
+@pytest.fixture
+def loader_function():
+    return DummyFunction()
+
+
+@pytest.fixture
+def preloaded(filenames, loader_function):
+    return vcm.Preloaded(filenames, loader_function)
+
+
+def test_loads_all_files(filenames, preloaded, loader_function):
```

Comment:
Here's another coupled fixture hiearchy. See https://github.com/VulcanClimateModeling/fv3net/pull/1112#discussion_r604544530.

I suggest turning the  `preloaded` fixture into a `get_preloaded` function to decouple these three fixtures.